dtfs-central-api/src/repositories/tfm-facilities-repo/tfm-facilities.repo.test.ts

```diff
@@ -0,0 +1,167 @@
+      '_id',
+      'dealId',
+      'type',
+      'hasBeenIssued',
+      'name',
+      'shouldCoverStartOnSubmission',
+      'coverStartDate',
+      'coverEndDate',
+      'issueDate',
+      'monthsOfCover',
+      'details',
+      'detailsOther',
+      'value',
+      'coverPercentage',
+      'interestPercentage',
+      'ukefFacilityId',
+      'dayCountBasis',
+    ];
+    it.each(REQUIRED_FACILITY_SNAPSHOT_FIELDS)("throws an error when the 'facilitySnapshot.%s' field is missing", (field) => {
+      // Arrange
+      const input = {
+        ...aValidTfmFacilityDocument(),
+        facilitySnapshot: {
+          ...aValidTfmFacilityDocumentFacilitySnapshotResult(),
+          [field]: undefined,
+        },
+      };
+
+      // Act / Assert
+      expect(() => TfmFacilitiesRepo.validateAndParseFindOneResult(input)).toThrow();
+    });
+
+    const NULLABLE_FACILITY_SNAPSHOT_FIELDS: (keyof ParsedTfmFacility['facilitySnapshot'])[] = ['coverStartDate', 'coverEndDate', 'issueDate', 'monthsOfCover'];
+    it.each(NULLABLE_FACILITY_SNAPSHOT_FIELDS)(
+      "does not throw an error when the 'facilitySnapshot.%s' field is null and returns the object with the same field",
+      (field) => {
+        // Arrange
+        const input = {
+          ...aValidTfmFacilityDocument(),
+          facilitySnapshot: {
+            ...aValidTfmFacilityDocumentFacilitySnapshotResult(),
+            [field]: null,
+          },
+        };
+
+        // Act / Assert
+        const result = TfmFacilitiesRepo.validateAndParseFindOneResult(input);
+        expect(result.facilitySnapshot[field]).toBeNull();
+      },
+    );
+
+    it('does not throw an error when the input contains an extra field and does not return than field in the result', () => {
+      // Arrange
+      const input = {
+        ...aValidTfmFacilityDocument(),
+        someOtherField: 'Some other value',
+      };
+
+      // Act / Assert
+      const result = TfmFacilitiesRepo.validateAndParseFindOneResult(input);
+      expect(result).not.toHaveProperty('someOtherField');
+    });
+
+    it("does not throw an error when the 'facilitySnapshot' object contains an extra field and does not return than field in the result", () => {
+      // Arrange
+      const input = {
+        ...aValidTfmFacilityDocument(),
+        facilitySnapshot: {
+          ...aValidTfmFacilityDocumentFacilitySnapshotResult(),
+          someOtherField: 'Some other value',
+        },
+      };
+
+      // Act / Assert
+      const result = TfmFacilitiesRepo.validateAndParseFindOneResult(input);
+      expect(result.facilitySnapshot).not.toHaveProperty('someOtherField');
+    });
+
+    const FACILITY_SNAPSHOT_DATE_FIELDS: (keyof ParsedTfmFacility['facilitySnapshot'])[] = ['coverStartDate', 'coverEndDate', 'issueDate'];
+    it.each(FACILITY_SNAPSHOT_DATE_FIELDS)("coerces the 'facilitySnapshot.%s.$date' field to a Date object when the value is an ISO string", (field) => {
```

Comment:
Do we want to coerce date fields from either an ISO string or a timestamp? Given they'll always be one of the types would it be clearer (and more useful) to only accept one of the types (in each case)

Reply:
This comment might not make sense how ive written it